source/Engine.h

```diff
@@ -102,6 +102,19 @@ class Engine {
 
 
 private:
+	class BufferIndex {
```

Comment:
I'm actually more fine with this name, but @quyykk still wants to not have a class for this.

Reply:
Typedef'd for the best of both worlds :)